src/pytest_mock_resources/fixture/database/relational/generic.py

```diff
@@ -154,6 +155,59 @@ def manage(self, session=None):
         finally:
             self.engine.dispose()
 
+    def manage_sync(self, session=None):
+        try:
+            self._run_actions()
+
+            if session:
+                if isinstance(session, sessionmaker):
+                    session_factory = session
+                else:
+                    session_factory = sessionmaker(bind=self.engine)
+
+                Session = scoped_session(session_factory)
+                session = Session(bind=self.engine)
+                yield session
+                session.close()
+            else:
+                yield self.engine
+        finally:
+            self.engine.dispose()
+
+    def manage_async(self, session=None):
```

Comment:
i feel like **this** function should be `async`, so that you can `await session.close()` after the session is yielded

Reply:
But isn't `self._run_actions` thus block the async code?

Reply:
It will, but so far as i know there's no mechanisms by which pytest tests are actually executed meaningfully concurrently in asyncio. All pytest-asyncio is doing is managing the event loop in, what are ultimately, synchronously and serially executed tests.

So blocking the event-loop with `self._run_actions` is really no different than `await`ing some theoretical async implementation of it, because there's only ever a single running coroutine on the loop at this point in the test setup.

Reply:
I see :ok_hand: